engines/wasmer/wasmer.go

```diff
@@ -95,10 +100,22 @@ type (
 var _ = (wapc.Module)((*Module)(nil))
 var _ = (wapc.Instance)((*Instance)(nil))
 
-var engineInstance = engine{}
+type EngineOption func(e *engine)
 
-func Engine() wapc.Engine {
-	return &engineInstance
+func Engine(opts ...EngineOption) wapc.Engine {
+	e := engine{}
+	for _, opt := range opts {
+		opt(&e)
+	}
+	return &e
+}
+
+func WithMetering(maxInstructions uint64, pfn unsafe.Pointer) EngineOption {
```

Comment:
if this change continues to poke holes for specific config instead of passing wasmer.Config, then we'll need to have docs on these functions. ex refer to PushMeteringMiddlewarePtr as otherwise it is hard for someone who doesn't know yet to understand what pfn means. Regardless, best to add an example test as show>tell

Reply:
Example is next, I promise!

Reply:
I hope the new godoc above was what was envisioned